src/Parallelization.php

```diff
@@ -516,4 +517,27 @@ private function runTolerantSingleCommand(
             }
         }
     }
+    
+        private function serializeInputOptions(InputInterface $input, array $blackListParams) : string {
+        $options = array_diff_key(
+            array_filter($input->getOptions()),
+            array_fill_keys($blackListParams, '')
+        );
+
+        $optionString = "";
+        foreach ($options as $name => $value) {
+            $definition = $this->getDefinition();
+            $option = $definition->getOption($name);
+            if (!$option->acceptValue()) {
+                $optionString .= ' --' . $name;
+            } elseif ($option->isArray()) {
+                foreach ($value as $arrayValue) {
+                    $optionString .= ' --'.$name.'='.$arrayValue;
+                }
+            } else {
+                $optionString .= ' --'.$name.'='.$value;
+            }
+        }
+        return $optionString;
```

Comment:
wouldn't it be simpler to return an array instead? We can deconstruct it above i.e. instead of having:

```
$this->serializeInputOptions($input, ['child','env','no-debug', 'processes']),

serializeInputOptions(): string
```

We can have:


```
...$this->serializeInputOptions($input, ['child','env','no-debug', 'processes']),

serializeInputOptions(): array // string[]
```

Reply:
I updated the code.